### DISCUSSION TITLE
njsPC not recognizing EasyTouch

### DISCUSSION THREAD

**Discussion by wsande** [tagyoureit/nodejs-poolController#912] [Q&A]
I am running njsPC and dashPanel on a Pi Zero 2W.  I have a USB-RS485 adapter plugged into the PI, and that is wired to the EasyTouch controller RS485 connector ( J20, Yellow, Green and Black).  DashPanel shows bytes being received on /dev/ttyUSB0, with Success.  It is Enabled, and Status is "Open"   For example, a few minutes after booting everything up, I see ~8000 bytes Received, with 464 Success.  

But DashPanel shows the controller type as "Nixie Single-body".  How do I set it (or get it to recognize) my EasyTouch controller?

In Settings -> Controller -> Model,  it says "Panel Type Nixie", and there is a drop-down for "Model", with just 4 types.  So it doesn't look like I can set EasyTouch manually.

I'm brand new to njsPC, and in fact this is the first RPi project I've tried.  

I did connect the RS-485 to my PC and look at what bytes are coming in.  A sample of bytes being received from the control panel looks like this:
00 00 ff 00 ff a5 00 10 60 04 01 ff 02 19 00 ff 
00 ff a5 00 10 60 06 01 04 01 20 00 00 00 ff 00 
ff a5 00 10 60 04 01 ff 02 19 00 ff 00 ff a5 00
10 60 06 01 04 01 20 00 00 ff 00 ff a5 00 10 60 
07 0f 04 02 02 00 00 00 00 00 00 00 00 00 01 00 00 01

Any pointers to whatever it is I'm missing here would be much appreciated!

Thanks,
ws

**Comment by tagyoureit** [comment]
These messages are valid RS485 messages coming from the pump to the controller.  But there are no messages from the Pentair controller in the short sample you provided.  You can use [messageManager](https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager) in dashPanel to view these.

Somehow, the EasyTouch packets are not coming through on the /dev/ttyUSB0 port.  Once njsPC sees a broadcast message from EasyTouch, it will automatically configure the application and not let you set up a Nixie controller.

**Comment by wsande** [comment]
MessageManager shows nothing at all.  I put a Logic analyzer on the RS485 and also monitored what is coming into /dev/ttyUSB0 using this command:
`od -x < /dev/ttyUSB0`

Which produces something like this:
```
poolpi@raspberrypi:~ $ od -x < /dev/ttyUSB0
0000000 0000 00ff a5ff 1000 0760 040f 0202 0000
0000020 0000 0000 0000 0100 2701 5c01 0000 ff00
0000040 ff00 00a5 6010 0104 02ff 0019 ff00 ff00
0000060 00a5 6010 0104 02ff 0019 00ff a5ff 1000
0000100 0660 0401 2001 0000 ff00 ff00 00a5 6010
0000120 0104 02ff 0019 00ff a5ff 1000 0660 0401
```
I do see valid sequences like   
`A5 60 10 01 04 02 19`
but many of them have extra FF or 00 in them:
`A5 60 10 01 04 FF 00 19`

1. Maybe I've connected the Rs-485 adapter to the wrong place on the EasyTouch control panel?  I have connected it to J20.
2. Or, is it possible my Easy Touch is too old or running the wrong F/W to work with njsPC?  It is an EasyTouch 8.

A pic is attached.  I'm powering the Pi from a +15V to +5V converter, where the +15 is also coming from J20.  That seems to work fine.
RS-485 adapter is connected to J20 YEL, GRN, and BLK - I guess BLK (GND) is optional for RS-485.  Going to the PI (which is a weatherproof enclosure outside the EasyTouch panel box, so it can connect to wifi) are the +5 and GND from the voltage converter, as well as the USB cable from the USB/RS-485 adapter.
 
![pool_Pi](https://github.com/tagyoureit/nodejs-poolController/assets/5423923/fb2305a1-ba61-4fe6-bf7e-529af5b06bb4)

**Reply by tagyoureit** [reply]
Can you run a [packet capture](https://github.com/tagyoureit/nodejs-poolController/wiki/How-to-capture-all-packets-for-issue-resolution#-dashpanel) and upload here.  You may have a filter (or not turned on the start button) on dashPanel that isn't showing the packets, but most likely they are there based on your 'od' command output.

**Reply by wsande** [reply]
I checked the connections again, and restarted everything (power cycled the panel and the RPi).

Now I am seeing messages in Message Manager.  All of them are  Source: Pump   Dest: OCP   

Packet capture is attached.  I ran it for a few minutes.

[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/14624706/replay.zip)

Any insight would be appreciated.

Thanks,
ws

**Comment by wsande** [comment]
Any insight from the attached packet capture?

**Reply by tagyoureit** [reply]
Sorry, been super busy.  I was hoping to see the application startup, but I don't think it would matter.  I'm still only seeing packets from your pump.  It's like looking at a highway but only seeing the eastbound traffic.  I would double check your wiring and make sure you don't have anything that's blocking communication in the other direction.  Maybe try removing a few of the other RS485 connections or make sure you don't have a break in one of the wires.
